proposals/12513-pvc-artifact-storage/README.md

```diff
@@ -0,0 +1,1867 @@
+
+**Acceptance Criteria:**
+
+- KFP works out-of-the-box with filesystem storage on local clusters
+- Artifacts are stored using the `kfp-artifacts://` URI scheme
+- No S3/GCS credentials required
+- Artifact viewing in UI works seamlessly
+
+#### Story 2: Operator Deploying KFP Without External Object Storage
+
+As an operator for a Kubeflow distribution, I want to deploy KFP with filesystem storage so that I don't need to productize and support a separate object storage system.
+
+**Acceptance Criteria:**
+
+- Single configuration option to enable filesystem storage
+- Artifact handling is part of KFP (no separate object storage component)
+- Storage automatically provisioned via PVCs
+- Backup/restore follows standard Kubernetes PVC procedures
+
+#### Story 3: Operator Configuring Storage Class and Size
+
+As an operator, I want to configure KFP to use a specific StorageClass and PVC size instead of defaults, so that I can match storage performance and capacity to my workload requirements.
+
+**Acceptance Criteria:**
+
+- Can specify `StorageClass` in KFP configuration
+- Can set PVC size limits (global configuration)
+- Storage quotas enforced via Kubernetes `ResourceQuotas`
+- Clear error messages when storage limits are reached
+- Can choose between RWO and RWX access modes based on needs
+
+#### Story 4: User Migrating from S3 to Filesystem Storage
+
+As a user with existing pipelines containing components that call `boto3.upload_file()` directly, I want KFP system artifacts to use `kfp-artifacts://` with PVC storage while my custom components continue accessing S3, so that I can migrate incrementally without rewriting all components at once.
+
+**Acceptance Criteria:**
+
+- KFP artifacts use filesystem storage via `kfp-artifacts://` URIs
+- Components using boto3/gsutil for S3 continue to work
+- Can mix storage backends (system artifacts on PVC, custom data on S3)
+- Clear documentation on what uses filesystem vs object storage
+
+#### Story 5: Operator Deploying Multi-Tenant KFP with Namespace Isolation
+
+As an operator, I want to deploy KFP in namespace-local mode where each namespace annotated with `pipelines.kubeflow.org/enabled=true` gets its own artifact server pod and dedicated PVC, so that Team A's artifacts in namespace `team-a` are physically isolated from Team B's artifacts in namespace `team-b`.
+
+**Acceptance Criteria:**
+
+- Each namespace with `pipelines.kubeflow.org/enabled=true` annotation gets its own artifact server deployment
+- Each namespace gets its own dedicated PVC (no shared storage)
+- Artifact server in `team-a` namespace cannot access PVC in `team-b` namespace
+- Users can only access artifacts in namespaces they have RBAC permissions for (via `SubjectAccessReview`)
+- Physical isolation verified: deleting `team-a` namespace doesn't affect `team-b`'s artifacts
```

Comment:
```suggestion
```

This storay has been made completely obsolete by the multi-tenant default seaweedfs. It would just be more complicated and break zero overhead namespaces. Its all done already in a secure manner and without massive overhead from additinal artifact servers per namespace. Just imagine 1000 namespaces and 1000 extra pods when idle.